src/Adapters/Symfony/EventListener/ErrorListener.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * This file is part of Collision.
+ *
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace NunoMaduro\Collision\Adapters\Symfony\EventListener;
+
+use Whoops\Exception\Inspector;
+use NunoMaduro\Collision\Provider;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+
+/**
+ * This is an Collision Symfony Adapter Error Listener implementation.
+ *
+ * @author Nuno Maduro <enunomaduro@gmail.com>
+ */
+class ErrorListener
+{
+    /**
+     * Retreives error from the provided event
+     * and ouputs details of that error to the event output.
+     *
+     * @param \Symfony\Component\Console\Event\ConsoleErrorEvent $event
+     */
+    public function onConsoleError(ConsoleErrorEvent $event)
+    {
+        $handler = (new Provider())->register()
+            ->getHandler()
+            ->setOutput($event->getOutput());
+
+        $handler->setInspector(new Inspector($event->getError()));
+
+        $handler->handle();
+
+        $event->setExitCode(0);
```

Comment:
why is the exit code changed?

Reply:
After checking the code https://github.com/symfony/console/blob/ac3d14fa82aad7ab067ecc84e0e1a4d01789cc76/Application.php#L233 I have seen that If I set the `exit code` as `0` the console application will terminate the execution without rendering the error on his traditional way.

Reply:
Hm, not a big fan but the alternative is overriding the Application `renderException()` method which is not really an option here. Maybe an alternative would be to put an `exit()` in the listener as well... Not sure

Reply:
@theofidry Hm, I think using an `exit()` on a listener is not a good option. Because maybe the dispatcher has another listener that want to listen to this event. Such as logger reporting, or other stuff.

Reply:
Yeah right, that wouldn't be a good idea